test/IISIntegration.FunctionalTests/Inprocess/ResponseHeaderTests.cs

```diff
@@ -51,15 +52,30 @@ public ResponseHeaders(IISTestSiteFixture fixture)
         }
 
         [ConditionalTheory]
-        [InlineData(200, "custom", "custom")]
-        [InlineData(500, "", "Internal Server Error")]
-        [InlineData(999, "", "")]
-        public async Task CustomErrorCodeWorks(int code, string reason, string expectedReason)
+        [InlineData(200, "custom", "custom", null)]
```

Comment:
you really don't need to add a test variation to cover it. Otherwise, we may end up with testing for every IIS API.